### PR TITLE
feat(import): resource name

### PR DESCRIPTION
**Pull request by BerkieBb**
## Description

Adds the ability to specify a resource name with the module loading

## Checklist

- [ ] I have personally loaded this code into an updated Qbox project and checked all of its functionality.
- [x] My code fits the style guidelines.
- [x] My PR fits the contribution guidelines.

### CODE REVIEW

File: import.lua [hunk 1]
```diff
@@ -5,22 +5,51 @@
...
+        'resourceName:client:module_name' -- The resource loading also accepts a side to load on
     }
 ]]
 
-local resourceName = 'qbx-core'
+local resources = {}
 local isServer = IsDuplicityVersion()
 
+local function getSide(value)
+    return value == 'client' and 'client' or value == 'server' and 'server' or value == 'shared' and 'shared' or 'none'
```
Comment (Manason) [inline import.lua:18]:
```suggestion
    if value == 'client' or value == 'server' or value == 'shared' then return value
    else return 'none' end
```

nit; I find this way to be a bit more readable.

File: import.lua [hunk 2]
```diff
@@ -5,22 +5,51 @@
...
+end
+
+for i = 0, GetNumResources() - 1 do
+    local resource = GetResourceByFindIndex(i)
+    if GetResourceState(resource) == 'started' then
+        resources[resource] = true
+    end
+end
+
 for i = 0, GetNumResourceMetadata(resourceName, 'module') - 1 do
```
Comment (Manason) [inline import.lua:32]: Is this function only called once to get the number of resource metadata? Or is it called on every iteration?

File: import.lua [hunk 3]
```diff
@@ -5,22 +5,51 @@
...
+    local resourceName = 'qbx-core'
     local module = GetResourceMetadata(resourceName, 'module', i)
     if module then
```
Comment (Manason) [inline import.lua:35]: Consider extracting a function and then inverting to use more guard clauses.

**Review by Manason — APPROVED** [review]

File: import.lua (hunk 2 above)
Comment (BerkieBb) [reply import.lua:32]: GetNumResourceMetadata will only be called once to get the number to reach until it needs to end

File: import.lua (hunk 1 above)
Comment (BerkieBb) [reply import.lua:18]: It seems readable enough for me as it is, pretty straight forward

**Review by Manason — APPROVED** [review]

**Pull request merged**